tests/modules/moduleconfigs.c

```diff
@@ -0,0 +1,136 @@
+#include "redismodule.h"
+#include <strings.h>
+int mutable_bool_val;
+int immutable_bool_val;
+long long longval;
+long long memval;
+RedisModuleString *strval;
+int enumval;
+
+/* Series of get and set callbacks for each type of config, these rely on the privdata ptr
+ * to point to the config, and they register the configs as such. Note that one could also just
+ * use names if they wanted, and store anything in privdata. */
+int getBoolConfigCommand(const char *name, void *privdata) {
+    REDISMODULE_NOT_USED(name);
+    return (*(int *)privdata);
+}
+
+int setBoolConfigCommand(const char *name, int new, void *privdata, const char **err) {
+    REDISMODULE_NOT_USED(name);
+    REDISMODULE_NOT_USED(err);
+    *(int *)privdata = new;
+    return REDISMODULE_OK;
+}
+
+long long getNumericConfigCommand(const char *name, void *privdata) {
+    REDISMODULE_NOT_USED(name);
+    return (*(long long *) privdata);
+}
+
+int setNumericConfigCommand(const char *name, long long new, void *privdata, const char **err) {
+    REDISMODULE_NOT_USED(name);
+    REDISMODULE_NOT_USED(err);
+    *(long long *)privdata = new;
+    return REDISMODULE_OK;
+}
+
+RedisModuleString *getStringConfigCommand(const char *name, void *privdata) {
+    REDISMODULE_NOT_USED(name);
+    REDISMODULE_NOT_USED(privdata);
+    return strval;
+}
+int setStringConfigCommand(const char *name, RedisModuleString *new, void *privdata, const char **err) {
+    REDISMODULE_NOT_USED(name);
+    REDISMODULE_NOT_USED(err);
+    REDISMODULE_NOT_USED(privdata);
+    size_t len;
+    /* Address sanitizer will complain about a leak if we're not properly freeing core side. */
+    if (!strcasecmp(RedisModule_StringPtrLen(new, &len), "rejectisfreed")) {
+        return REDISMODULE_ERR;
+    }
```

Comment:
i don't understand this comment, but it smells like it's covering a bug.
the code looks valid, i.e. if we get a value that we wanna reject, we return an error, and keep (avoid freeing) the old pointer.
so maybe the comment about the leak is outdated?

p.s. let's return some error string in `*err` to test that infra.